web/src/lib/components/pages.tsx

```diff
@@ -125,8 +125,15 @@ export default class Pages extends Component<PagesProps, PagesState> {
     }
   }
 
+  /**
+   * If the iOS app is installed, open it. Otherwise, open the App Store.
+   */
   private openInApp() {
-    window.location.href = getItunesURL();
+    window.location.href = 'commonvoice://';
```

Comment:
This pops up an error message on iOS Firefox saying it doesn't understand the url. Is there a way we can feature detect for custom protocols? I googled it but didn't find anything. A backup would be to add a `Utility.isSafari` check here before setting this url.

Reply:
You can't feature detect custom protocols, because you can't know in the browser which apps the user has installed. We need this to work in all browsers except Safari. Those browsers are also based on Safari on iOS. Apple doesn't like this kind of custom links because they want us to use [Universal Links](https://developer.apple.com/ios/universal-links/). Previously, a workaround for the popup was to put the custom link in an iframe, but they fixed that 'bug' as well, so now there's always such a popup if you use your own links.

We could maybe use Universal Links, but it involves generating one in an App Store developer account so I can't really do much about that. When using Universal Links, it's impossible for other websites to open your app with a similar link.

Reply:
Ah ok, good to know and thanks for the background. Overall the experience is nicer with your change than without it, since they should only see that error message once, and then once updated always go right to the app. Let's keep it.

Reply:
For the time being, we're going to disable the custom protocol functionality on the site, but still register the custom protocol in the app. In the future, if we are assured more users have updated the app, we can add back in this improved redirect flow.